solutions/src/main/java/ru/sberbank/school/task02/ExtendedConverter.java

```diff
@@ -26,52 +27,52 @@ public ExtendedConverter(ExternalQuotesService externalQuotesService) {
 
         List<Quote> quotes = externalQuotesService.getQuotes(symbol);
 
-        Quote quote = null;
-        Quote reserveQuote = null;
+        List<Quote> quotesResult = new ArrayList<>();
 
         for (Quote currQuote : quotes) {
             BigDecimal price = getPrice(operation, currQuote);
-            BigDecimal usd = amount.divide(price, 15, BigDecimal.ROUND_HALF_UP);
-
-            if (currQuote.getVolumeSize().compareTo(usd) > 0 || currQuote.isInfinity()) {
-                Quote quoteUsd = searchQuote(usd, quotes);
-                if (currQuote.getVolumeSize().compareTo(quoteUsd.getVolumeSize()) == 0) {
-                    if (quote == null) {
-                        quote = quoteUsd;
-                    } else {
-                        reserveQuote = quoteUsd;
-                    }
+            BigDecimal amountBaseCurrency = amount.divide(price, 10, BigDecimal.ROUND_HALF_UP);
+
+            if (currQuote.getVolumeSize().compareTo(amountBaseCurrency) > 0 || currQuote.isInfinity()) {
+                Quote quoteBaseCurrency = searchQuote(amountBaseCurrency, quotes);
+                if (currQuote.getVolumeSize().compareTo(quoteBaseCurrency.getVolumeSize()) == 0) {
+                    quotesResult.add(currQuote);
                 }
             }
         }
 
-        if (quote == null) {
+        if (quotesResult.isEmpty()) {
             return Optional.empty();
         }
 
-        if (reserveQuote != null) {
-            quote = choiceBeneficiary(quote, reserveQuote, beneficiary);
-        }
-
-        BigDecimal price = getPrice(operation, quote);
+        BigDecimal price = choicePriceByBeneficiary(quotesResult, beneficiary, operation);
 
-        return Optional.ofNullable(BigDecimal.ONE.divide(price, 15, BigDecimal.ROUND_HALF_UP));
+        return Optional.ofNullable(BigDecimal.ONE.divide(price, 10, BigDecimal.ROUND_HALF_UP));
 
     }
 
-    private Quote choiceBeneficiary(Quote quote, Quote reserveQuote, Beneficiary beneficiary) {
-        if (reserveQuote.isInfinity()) {
-            return beneficiary == Beneficiary.CLIENT ? reserveQuote : quote;
-        }
+    private BigDecimal choicePriceByBeneficiary(List<Quote> quotes, Beneficiary beneficiary, ClientOperation operation) {
```

Comment:
Line is longer than 120 characters (found 121).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-33684087>*